crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -170,4 +170,72 @@ def cancel
   def repocheck
     render json: Api::Upgrade.repocheck
   end
+
+  protected
+
+  api :POST, "/api/upgrade/new",
+    "Initialization of Crowbar during upgrade with creation of a new database.
```

Comment:
It sounds like the database creation is part of this endpoint.
If that's not the case, please update the description